Add notes from 2022-07-11 meeting (#2)
bytecodealliance/meetings: cranelift/2022/cranelift-07-11.md

  * working on understanding the state of fuzzing in wasmtime/cranelift
  * evaluating where fuzzing gaps exist
  * are you interested in fixing issue #3347, afonzo?
    * afonzo: i'd prefer if you could take that; the interpreter is being
      developed to aid fuzzing
* anton:
  * alex crichton reported an issue with constant generation in the aarch64
    backend
  * refactoring/improving the aarch64 backend
* trevor:
  * started at fastly a little over a month ago with the wasm team
  * working on finishing the x64 backend isle migration
* nick:
  * improving performance for stacktrace generation: the slowdown seems linear
    in the number of modules loaded
  * changing codegen to omit frame pointers for leaf functions that don't trap
    * walking over the whole clif function semes expensive, is there a better
      approach here?
* johnnie:
  * pushed the patch for a self-hosted runner to help with automated
    benchmarking
  * current state is that it only runs benchmarks for the x64 backend
  * we can expand to aarch64 later

#### Determining if clif functions trap

* nick: can we use the trap table for a function to avoid walking the whole
  function?

> * afonzo: can we keep some metadata while we're generating the function that we

Reviewer: ```suggestion
* anton: can we keep some metadata while we're generating the function that we
```

  set when generating instructions that can trap?
* nick: this semes far away from where the information would be used; what if
  the code that could trap becomes dead after optimization passes?

> * afonzo: can we introduce traps through optimizations?

Reviewer: ```suggestion
* anton: can we introduce traps through optimizations?
```

  * nick: yes, through legalization
* nick: i'll investigate the trap table approach, and fall back on modifying the
  instruction builder to record trap metadata if that doesn't work

#### Performance impact of preserving frame pointers

* afonzo: leaf functions that trap will save the frame pointer?
  * nick: yes exactly
* jamey: is there a flag for enabling debugging symbols that we could turn off
  for optimized builds?
* nick: there are flags available, but nothing specific that we might use to
  control preserving the frame pointer

#### How would fp preservation interact with authenticated pointers?

* nick: it shouldn't affect authenticated pointers

> * afonzo: the upper 16 bits would be used for the signature

Reviewer: ```suggestion
* anton: the upper ~16 bits (configurable) would be used for the signature
```
